test/src/main/java/org/apache/accumulo/test/RootRecoveryIT.java

```diff
@@ -0,0 +1,119 @@
+import org.apache.hadoop.conf.Configuration;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Iterables;
+
+public class RootRecoveryIT extends SharedMiniClusterBase {
+
+  private static final int ZK_TIMEOUT_MS = 5000;
+  private static final Logger LOG = LoggerFactory.getLogger(RootRecoveryIT.class);
+
+  private static class RootRecoveryITConfigurationCallback
+      implements MiniClusterConfigurationCallback {
+    @Override
+    public void configureMiniCluster(MiniAccumuloConfigImpl cfg, Configuration coreSite) {
+      cfg.setProperty(Property.INSTANCE_ZK_TIMEOUT, ZK_TIMEOUT_MS + "ms");
+    }
+  }
+
+  @Override
+  protected Duration defaultTimeout() {
+    return Duration.ofMinutes(3);
+  }
+
+  @BeforeAll
+  public static void start() throws Exception {
+    SharedMiniClusterBase.startMiniClusterWithConfig(new RootRecoveryITConfigurationCallback());
+  }
+
+  @AfterAll
+  public static void stop() throws Exception {
+    SharedMiniClusterBase.stopMiniCluster();
+  }
+
+  @Test
+  public void testRootRecovery() throws Exception {
+
+    // Insert some delete mutations into the metadata table
+    try (BatchWriter writer =
+        getCluster().getServerContext().createBatchWriter(SystemTables.METADATA.tableName())) {
+      var mutation = new Mutation("3;<");
+      mutation.putDelete(TabletColumnFamily.AVAILABILITY_COLUMN.getColumnFamily(),
+          TabletColumnFamily.AVAILABILITY_COLUMN.getColumnQualifier());
+      writer.addMutation(mutation);
+      writer.flush();
+    } catch (TableNotFoundException | MutationsRejectedException e) {
+      throw new RuntimeException(e);
+    }
+
+    // Compact the metadata table, which will end up writing mutations
+    // to the root tablet for new metadata files
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+      c.tableOperations().compact(SystemTables.METADATA.tableName(), null, null, true, true);
+    }
+
+    getCluster().getClusterControl().stopAllServers(ServerType.MANAGER);
+    getCluster().getClusterControl().stopAllServers(ServerType.TABLET_SERVER);
+
+    // Let the ZooKeeper locks expire
+    Thread.sleep(ZK_TIMEOUT_MS * 2);
+
+    getCluster().getClusterControl().startAllServers(ServerType.TABLET_SERVER);
+    getCluster().getClusterControl().startAllServers(ServerType.MANAGER);
+
+    // This scan should not time out
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+      Wait.waitFor(() -> c.instanceOperations().getServers(Type.MANAGER).size() == 1);
+      Wait.waitFor(() -> c.instanceOperations().getServers(Type.TABLET_SERVER).size() == 2);
+
+      LOG.info("Scanning root table");
+      Scanner s = c.createScanner(SystemTables.ROOT.tableName());
+      @SuppressWarnings("unused")
+      int ignoredSize = Iterables.size(s);
```

Comment:
I think spotbugs will flag on Iterables here. Can also suppress the warning with a more specific assertion on the returned size.

```suggestion
      LOG.info("Scanning root table");
      try (Scanner s = c.createScanner(SystemTables.ROOT.tableName())) {
        assertTrue(s.stream().count() > 0);
      }
```

Reply:
It doesn't look like SpotBugs failed on this.

Reply:
It's also nice to avoid unnecessary 3rd party libraries when we can do the same thing with a built-in Java construct that is just as easy to read.